src/promptflow/promptflow/_sdk/operations/_run_operations.py

```diff
@@ -293,7 +298,20 @@ def _visualize(self, runs: List[Run], html_path: Optional[str] = None) -> None:
             )
             details.append(copy.deepcopy(detail))
             metadatas.append(asdict(metadata))
-        data_for_visualize = RunVisualization(detail=details, metadata=metadatas)
+            # TODO: add language to run metadata
+            flow_dag = yaml.safe_load(metadata.dag)
+            configs.append(
+                VisualizationConfig(
+                    [AvailableIDE.VS_CODE]
+                    if flow_dag.get(LANGUAGE_KEY, FlowLanguage.Python) == FlowLanguage.Python
+                    else [AvailableIDE.VS]
```

Comment:
ideally the logic should be: 
 request triggered from vs exetension are targeting vs, not all csharp flow target vs

Reply:
hmm, that will require a further discussion with executor & vs extension on how to pass the information.

Let's track it with another item and merge this PR first: [Task 2835185](https://msdata.visualstudio.com/Vienna/_workitems/edit/2835185): tell sdk where the batch run is triggered to optimize pf run visualize